torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -2896,6 +2896,39 @@ struct to_ir {
         }
         return iterable_tree;
       }
+      case prim::list: {
+        if (apply.inputs().size() == 0) {
+          TypePtr type = type_hint ? type_hint : ListType::ofTensors();
+          return std::make_shared<SimpleValue>(graph->insertNode(graph->createList(type, {}))->output());
+        }
+        // list(iter) desugars to [_elem for _elem in iter]
+        checkApplyNumInputs(apply, 1);
+        auto iter_input = emitSugaredExpr(apply.inputs()[0], 1);
+
+        // aten::list builtin op is registered for List and Str input
+        // dispatch to the builtin op to avoid perf slowdown on existing uses
+        if (auto simple = asSimple(iter_input)) {
+          if (simple->type()->cast<ListType>() || simple->type()->cast<StringType>()) {
+            return std::make_shared<SimpleValue>(emitBuiltinCall(
+              apply.range(), *method.graph(), aten::list, {simple}, {}));
+          }
+        }
+        const std::string& iter_name = createTempName("$_iter");
+        environment_stack->setSugaredVar(
```

Comment:
Should this be removed from the environment stack after emitting the list comprehension?

Reply:
It's not really necessary, `createTempName` creates a unique identifier which won't ever be used again, and `$_iter` is not user-valid variable name. In the other instances of createTempName we do not subsequently remove the temporary variable binding